scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -106,6 +108,41 @@ private Optional<JsonObject> postRequest(final String gqlStr) throws IOException
     }
   }
 
+  private Optional<JsonObject> postRequest(final String query, final JsonObject variables) throws IOException, PlanServiceException {
+    try {
+      //TODO: (mem optimization) use streams here to avoid several copies of strings
+      final var reqBody = Json
+          .createObjectBuilder()
+          .add("query", query)
+          .add("variables", variables)
+          .build();
+      final var httpReq = HttpRequest
+          .newBuilder().uri(merlinGraphqlURI).timeout(httpTimeout)
+          .header("Content-Type", "application/json")
+          .header("Accept", "application/json")
+          .header("Origin", merlinGraphqlURI.toString())
+          .POST(HttpRequest.BodyPublishers.ofString(reqBody.toString()))
+          .build();
+      //TODO: (net optimization) gzip compress the request body if large enough (eg for createAllActs)
+      final var httpResp = HttpClient
+          .newHttpClient().send(httpReq, HttpResponse.BodyHandlers.ofInputStream());
+      if (httpResp.statusCode() != 200) {
+        //TODO: how severely to error out if aerie cannot be reached or has a 500 error or json is garbled etc etc?
+        return Optional.empty();
+      }
+      final var respBody = Json.createReader(httpResp.body()).readObject();
+      if (respBody.containsKey("errors")) {
+        throw new PlanServiceException(respBody.toString());
+      }
+      return Optional.of(respBody);
+    } catch (final InterruptedException e) {
```

Comment:
We should at least reset the interrupted flag here (Thread.currentThread().interrupt())

My limited understanding is that InterruptedException is thrown when the thread receives a request to terminate (ctrl+c or similar...?), so I don't think we'll want to retry in that scenario.

Reply:
I 100% copy pasted the new request with arguments method from the existing request by string method, so I will 100% defer to you on this element and happy to make the change in both methods